File: MyFullProject/spring-backend/springbackend/src/main/java/com/example/springbackend/dto/MessageDto.java
package com.example.springbackend.dto;

import java.time.LocalDateTime;

import lombok.Getter;
import lombok.Setter;
@Getter
@Setter

public class MessageDto {

    private Long id;
    private Long senderId;
    private String senderName;
    private Long profileImageId;

    private String content;
    private String status;
    private LocalDateTime sentAt;

    private Long replyToMessageId;
    private String replyPreview;
    private String replySenderName;

    // getters setters
}

File: MyFullProject/android-app/MyApplication/app/src/main/java/com/example/myapplication/models/MessageDto.kt
package com.example.myapplication.models

data class MessageDto(

    val id: Long,

    val senderId: Long,

    val senderName: String,

    val senderProfileImageId: Long?,

    val content: String,

    val status: String,

    val sentAt: String,

    val replyToMessageId: Long?,

    val replyPreview: String?,

    val replySenderName: String?
)